allauth/socialaccount/providers/soundcloud/provider.py

```diff
@@ -16,7 +16,7 @@ def to_str(self):
         dflt = super(SoundCloudAccount, self).to_str()
         full_name = self.account.extra_data.get("full_name")
         username = self.account.extra_data.get("username")
-        return full_name or username or dflt
+        return username or full_name or dflt
```

Comment:
This could be folded into the base `to_str()`.

Reply:
The base `to_str` does not look at `full_name`, which may be needed here as a fallback.